packages/docusaurus/bin/beforeCli.mjs

```diff
@@ -16,16 +16,24 @@ import boxen from 'boxen';
 import {createRequire} from 'module';
 
 const packageJson = createRequire(import.meta.url)('../package.json');
-const sitePkg = createRequire(path.join(process.cwd(), 'package.json'))(
-  './package.json',
-);
+/** @type {Record<string, any>} */
+let sitePkg;
+try {
+  sitePkg = createRequire(path.resolve('package.json'))('./package.json');
+} catch {
+  logger.warn`path=${'package.json'} file not found at CWD: path=${process.cwd()}.`;
+  logger.info`This is non-critical, but could lead to undesired behavior downstream. Docusaurus assumes that path=${'package.json'} exists at CWD, because it's where the package manager looks up the script at. A common reason is because you have changed directory in the script. Instead of writing code=${'"start": "cd website && docusaurus start"'}, consider using the code=${'[siteDir]'} argument: code=${'"start": "docusaurus start website"'}.`;
+  sitePkg = {};
+}
 
 const {
   name,
   version,
   engines: {node: requiredVersion},
 } = packageJson;
 
+process.env.DOCUSAURUS_VERSION = version;
```

Comment:
I'd prefer to keep static requires instead of relying of this mysterious and implicit env var.

Can't we find a place to export this version?

Reply:
The `beforeCli` is purely side-effect and doesn't return anything. I don't really want to change that fact.

The env is very portable and it's internal so doesn't need to be "explicit". A lot of tools will put their version in the env. An alternative is to register in `process.versions` instead, but afaict that's mostly used for runtime dependencies, so less appropriate for us.

Reply:
But can't we just read that version once and expose it in a more explicit way?

What prevents doing this instead?


```js
import {DOCUSAURUS_VERSION} from '@docusaurus/utils';
```

Utils is supposed to have the same version as core (or you can add it as peerDeps)

Reply:
Yeah, I'm more worried that in the long term utils may not be the same version of core, otherwise I would have done it already.

Reply:
if that becomes the case we can add core as peerDeps, and/or move constants to a separate package

For now that seems good enough and better than an implicit env variable